src/libraries/System.Private.CoreLib/src/System/Text/EncodingTable.cs

```diff
@@ -119,6 +120,26 @@ internal static EncodingInfo[] GetEncodings()
             return arrayEncodingInfo;
         }
 
+        internal static EncodingInfo[] GetEncodings(Dictionary<int, EncodingInfo> encodingInfoList)
+        {
+            Debug.Assert(encodingInfoList != null);
+
+            for (int i = 0; i < s_mappedCodePages.Length; i++)
```

Comment:
It'd be better to store s_mappedCodePages into a local and then use that local everywhere.